File: 944-smallest-range-i/smallest-range-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var smallestRangeI = function(nums, k) {

    let maxN = Math.max(...nums)-k;
    let minN = Math.min(...nums)+k;

    let res = maxN-minN;

    if(res<=0) return 0;
    else return res;
    
};

File: 944-smallest-range-i/smallest-range-i.java
class Solution {
    public int smallestRangeI(int[] nums, int k) {
        int min = nums[0];
        int max = nums[0];
        int res = 0;
        
        for (int x : nums) {
            min = Math.min(min, x);
            max = Math.max(max, x);
        }
        
        res = (max-k)-(min+k);
        
        return Math.max(0, res);
    }
}
